smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/KafkaConsumerRebalanceListener.java

```diff
@@ -0,0 +1,37 @@
+package io.smallrye.reactive.messaging.kafka;
+
+import java.util.Set;
+
+import io.vertx.kafka.client.common.TopicPartition;
+import io.vertx.mutiny.kafka.client.consumer.KafkaConsumer;
+
+/**
+ *
+ * When implemented by a managed bean annotated with {@link javax.inject.Named} and
+ * configured against a consumer, ex:
+ * mp.messaging.incoming.example.consumer-rebalance-listener.name=ExampleConsumerRebalanceListener
+ *
+ * Will be applied as a consumer rebalance listener to the consumer.
+ *
+ * For more details
+ *
+ * @see org.apache.kafka.clients.consumer.ConsumerRebalanceListener
+ */
+public interface KafkaConsumerRebalanceListener {
```

Comment:
What's the reason for not using `org.apache.kafka.clients.consumer.ConsumerRebalanceListener` directly?

Reply:
And then I read the next file and understood it. Sorry!

Reply:
For documentation's sake it's because most of the time you'll want to manipulate the consumer object (like seeking to an offset). The original listener interface is designed with an imperative synchronous style in mind https://kafka.apache.org/0100/javadoc/org/apache/kafka/clients/consumer/ConsumerRebalanceListener.html

I originally wrote it with the the vanilla listener interface then saw the problem 😆

With an example in the docs it'll be clearer.